src/Misc/PartitionWithCondition.jl

```diff
@@ -0,0 +1,104 @@
+@doc raw"""
+  partition_with_condition(
```

Comment:
Would you please move this docstring to the actual function `partition_with_condition` in line 50?
Then if someone types `?partition_with_condition` they will see the docstring.